src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,12 +1,38 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import { Loader } from '../Loader';
+import { Todo } from '../../types/Todo';
+import { User } from '../../types/User';
+import { getUser } from '../../api';
+
+interface Props {
+  selectedTodo: Todo;
+  onSelectedTodo: (todo: Todo | undefined) => void;
+  onModalOpened: (isOpen: boolean) => void;
+}
+
+export const TodoModal: React.FC<Props> = ({
+  selectedTodo,
+  onSelectedTodo,
+  onModalOpened,
+}) => {
+  const [user, setUser] = useState<User>();
+
+  useEffect(() => {
+    if (selectedTodo.userId) {
+      getUser(selectedTodo.userId).then(usr => setUser(usr));
```

Comment:
Avoid directly setting the state with props inside useEffect. Instead, consider using a function to derive the state from props if necessary. This can prevent potential issues with stale state values.